articles/transfer-learning-with-deep-learning/index.md

```diff
@@ -0,0 +1,121 @@
+
+Domain adaptation can be categorized as either homogenous or heterogeneous. This is dependent on how homogenous the feature spaces both source and target domains are. A feature space is a set of features that are identified from data. Homogenous domain adaptation refers to where two domains have the same feature spaces but different feature distributions. Heterogeneous domain adaptation has heterogeneous feature spaces in different domains. As a result, it is much more difficult to take on. The feature spaces have no similarity. To link up two different domains, supplementary information, such as labels, is required. If such information is not available, it is hard to deal with heterogeneous domain adaptation. The unavailability of supplementary information is a common scenario in practice.
+
+For further reading, I mentioned a couple of applications involving domain adaptation in this [article](https://www.section.io/engineering-education/basics-of-transfer-learning/).
+
+#### Domain Confusion
+
+Consider feature representation transfer, which is covered in my previous transfer learning [article](https://www.section.io/engineering-education/basics-of-transfer-learning/). We defined it as basically “transferring features” from source to target domains. In a neural network, different layers identify different sets of features. Such an approach can be used to learn features that do not vary regardless of the domain. This helps improve their transferability across the domains. In such a context, the feature representations between the source and target domains are made to be as similar as possible. This is done at the expense of allowing the model to learn any feature representations.
+
+But why? The goal is to add an objective to the model at the source to encourage similarity with the target by confusing the source domain itself. Specifically, domain confusion loss is used to confuse the high-level classification layers of a neural network by matching the distributions of the target and source domains.
+
+A fitting example of such a network is a Domain-Adversarial Neural Network (DANN). The network and relevant concepts around it are expressed in the paper [Domain-Adversarial Training of Neural Networks](https://arxiv.org/abs/1505.07818). DANNs consist of two losses; classification loss and domain confusion loss. The aim is to make sure samples come across as mutually indistinguishable to the classifier. To achieve this, one has to minimize the classification loss for the source samples. One has to also minimize the domain confusion loss for all samples.
+
+#### Multitask Learning
```

Comment:
Can you give an example here? Including a link to your article on GPT-3 would make sense.

Reply:
Also, only after reading your article did I realize GPT-3 is a zero-shot learning algorithm. Can you differentiate between the two? How is multi-task learning different from GPT-3? The question arises because GPT-3 is a multi-task model.